library/alloc/src/collections/binary_heap.rs

```diff
@@ -118,19 +121,16 @@
 //!     // Chosen for its efficiency.
 //!     let graph = vec![
 //!         // Node 0
-//!         vec![Edge { node: 2, cost: 10 },
-//!              Edge { node: 1, cost: 1 }],
+//!         vec![Edge { node: 2, cost: 10 }, Edge { node: 1, cost: 1 }],
 //!         // Node 1
 //!         vec![Edge { node: 3, cost: 2 }],
 //!         // Node 2
-//!         vec![Edge { node: 1, cost: 1 },
-//!              Edge { node: 3, cost: 3 },
-//!              Edge { node: 4, cost: 1 }],
+//!         vec![Edge { node: 1, cost: 1 }, Edge { node: 3, cost: 3 }, Edge { node: 4, cost: 1 }],
```

Comment:
It looks like this is using rustc's rustfmt settings including `use_small_heuristics = "Max"`. I think default settings would be more appropriate for library docs. But maybe that's getting too picky.

Reply:
It might be tricky to do that without splitting the code blocks into their own files and include them (which I'm happy to try, if that would be an acceptable solution).  An alternate solution would to instead add rustfmt file(s) for the rust library directories (but then it's applied across the libraries outside of doc comments).  I don't think having specific format settings for docs is something that's even planned in rustfmt.

Reply:
> It might be tricky to do that without splitting the code blocks into their own files and include them (which I'm happy to try, if that would be an acceptable solution).

That would make updating documentation along with functions more tedious and easy to miss. Some macro-generated code already contains lots of annotations, an include could be easily missed.

> An alternate solution would to instead add rustfmt file(s) for the rust library directories (but then it's applied across the libraries outside of doc comments).

That'd lead to reformatting all of the library code, no?

> I don't think having specific format settings for docs is something that's even planned in rustfmt.

Well, _if_ we decide that doc comment examples need different formatting then something like that might be necessary.